src/storage/mvcc/reader/scanner/forward.rs

```diff
@@ -417,6 +399,132 @@ impl<S: Snapshot> ScanPolicy<S> for LatestKvPolicy {
     }
 }
 
+/// The ScanPolicy for outputting `TxnEntry`.
+///
+/// The `ForwardScanner` with this policy only outputs records whose commit_ts
+/// is greater than `after_ts`. It also supports outputting delete records
+/// if `output_delete` is set to `true`.
+pub struct LatestEntryPolicy {
+    after_ts: TimeStamp,
+    output_delete: bool,
+}
+
+impl LatestEntryPolicy {
+    pub fn new(after_ts: TimeStamp, output_delete: bool) -> Self {
+        LatestEntryPolicy {
+            after_ts,
+            output_delete,
+        }
+    }
+}
+
+impl<S: Snapshot> ScanPolicy<S> for LatestEntryPolicy {
+    type Output = TxnEntry;
+
+    fn handle_lock(
+        &mut self,
+        current_user_key: Key,
+        cfg: &mut ScannerConfig<S>,
+        cursors: &mut Cursors<S>,
+        statistics: &mut Statistics,
+    ) -> Result<HandleRes<Self::Output>> {
+        scan_latest_handle_lock(current_user_key, cfg, cursors, statistics)
+    }
+
+    fn handle_write(
+        &mut self,
+        current_user_key: Key,
+        cfg: &mut ScannerConfig<S>,
+        cursors: &mut Cursors<S>,
+        statistics: &mut Statistics,
+    ) -> Result<HandleRes<Self::Output>> {
+        // Now we must have reached the first key >= `${user_key}_${ts}`. However, we may
+        // meet `Lock` or `Rollback`. In this case, more versions needs to be looked up.
+        let mut write_key = cursors.write.key(&mut statistics.write);
+        let entry: Option<TxnEntry> = loop {
+            if Key::decode_ts_from(write_key)? <= self.after_ts {
+                // There are no newer records of this key since `after_ts`.
+                break None;
+            }
+            let write_value = cursors.write.value(&mut statistics.write);
+            let write = WriteRef::parse(write_value)?;
+            statistics.write.processed += 1;
+
+            if write.write_type == WriteType::Put
```

Comment:
can you write it in `match`? seems we can separate`Put` with `Delete` to make the logic more clear.